pylint/checkers/strings.py

```diff
@@ -787,3 +871,66 @@ def str_eval(token):
     if token[0:3] in ('"""', "'''"):
         return token[3:-3]
     return token[1:-1]
+
+
+def is_long_string(token: str) -> bool:
+    """Is this token a "longstring" (is it well-formed and triple-quoted)?
+
+    Long strings are triple-quoted as defined in
+    https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals
+
+    Args:
+        token: The token to be parsed.
+
+    Returns:
+        A boolean representing whether or not this token matches a longstring
+        regex.
+    """
+    single_quoted_regex = "(%s)?'''.*'''\Z" % "|".join(_STRING_AND_BYTES_PREFIXES)
```

Comment:
do you need to check the complete string (as opposed to just the prefix) here?

Reply:
Mmm.  You're totally right.  The only thing we get from checking the rest of the string is some extra validation that the token is well-formed.

But checking well-formedness isn't really this function's job, and per your point, the extra code obscures what the function's purpose is.  So I'm going to make the change and add a little description in the docstring about well-formedness non-guarantees.

Reply:
Ok, now I'm second-guessing this decision.  The function isn't `has_long_string_prefix`, it claims to be `is_long_string.`  I want a function that purports to make that sort of judgment check for well-formedness.

So maybe the next question is to ask whether the implemented functionality actually needs an `is_long_string` function, and maybe if it should call a `has_long_string_prefix` instead.  There... I dunno.  I think if we need to write code anyway, then writing a function like `is_long_string` is both *very* slightly more useful and a little easier to mentally chunk.

So I think my inclination is to leave this code as-is.  What are your thoughts?

Reply:
my take is that a string that is not well-formed will not pass the python compiler, and therefore we can assume that a string starting with `'''` will necessarily end with `'''`. so keeping the function as `is_long_string` but only checking the prefix is fine here.

Reply:
Ok.  I think that's a good argument.  I'll go in and make the change.